xdsl/dialects/onnx.py

```diff
@@ -26,6 +26,54 @@
 from xdsl.utils.exceptions import VerifyException
 
 
+def verify_shapes(
+    lhs_type: TensorType[Attribute] | list[int],
+    rhs_type: TensorType[Attribute] | list[int],
+    res_type: TensorType[Attribute],
+) -> None:
+    """
+    https://github.com/onnx/onnx/blob/main/docs/Broadcasting.md
+    """
+
+    # Check that the arguments are broadcastable (using Numpy semantics) and that the result type is correct.
+    res_shape: list[int] = []
+    if isinstance(lhs_type, TensorType):
+        lhs_shape = lhs_type.get_shape()
+    else:
+        lhs_shape = lhs_type
+
+    if isinstance(rhs_type, TensorType):
+        rhs_shape = rhs_type.get_shape()
+    else:
+        rhs_shape = rhs_type
+    # Iterate over the shapes in reverse order and compute the result shape.
+    i = max(len(lhs_shape), len(rhs_shape))
+    while i > 0:
```

Comment:
Can this become a bit more Pythonic? Not fond of `while` and `continue` statements if they can be avoided.

Reply:
see @superlopuh's suggestion below